pkg/controller/machinepool/vsphereactuator.go

```diff
@@ -90,6 +90,7 @@ func (a *VSphereActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool
 							Folder:         setFolderPath(cd.Spec.Platform.VSphere.Folder, cd.Spec.Platform.VSphere.Datacenter, logger),
 							ComputeCluster: setComputeClusterPath(cd.Spec.Platform.VSphere.Cluster, cd.Spec.Platform.VSphere.Datacenter, logger),
 							Networks:       []string{cd.Spec.Platform.VSphere.Network},
+							Template:       a.osImage,
```

Comment:
One more change needed:
```
							Datacenter:     cd.Spec.Platform.VSphere.Datacenter,
```
(first thing under Topology, if you care to match the order of the struct def)